File: Day 9/0 Two Sum/solution.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        
        //
        Dictionary<int,int> map =  new  Dictionary<int,int>();
            
            //   1
        for (int i=0; i<nums.Length; i++){
            
            // 2              9    -    7
            var requiredNum = target - nums[i];
            
            //if the required num is there in the map, return the indexes of curent num and required num
            if(map.ContainsKey(requiredNum)){

                return new int[]{map[requiredNum], i};
            }
            
            //adding to map
            if(!map.ContainsKey(nums[i])){
                map.Add(nums[i],i);   // 2 > 0 | 
            }
        }
        
        return new int[]{};
        
    }
} 

File: Day 9/0 Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i in range(len(nums)):
            requiredNum = target - nums[i]
            if requiredNum in map:
                return [map[requiredNum], i]
            if nums[i] not in map:
                map[nums[i]] = i
        return []
